rpcs3/rpcs3qt/game_list_frame.cpp

```diff
@@ -2725,9 +2754,12 @@ void game_list_frame::PopulateGameGrid(int maxCols, const QSize& image_size, con
 	// Get list of matching apps
 	QList<game_info> matching_apps;
 
+	// Fallback is not needed when at least one entry is visible
+	const bool use_search_fallback = std::none_of(m_game_data.begin(), m_game_data.end(), [this](auto& game){ return IsEntryVisible(game); });
```

Comment:
I think this can be optimized.
Push to matching_apps here as well and just skip the lower loop if use_search_fallback is false